### ISSUE TITLE
CI Test (gtest) failing on Windows

### ISSUE THREAD

**Issue by pattacini** [robotology/icub-main#967]
### Bug description

Today, all of a sudden, CI started reporting failures during testing (using `googletest`) only on Windows (SHARED_LIBS:OFF).

Here's the generic error:
```console
unknown file: error: SEH exception with code 0xc0000005 thrown in the test body.
```

### Steps to reproduce

See:
- https://github.com/robotology/icub-main/actions/runs/9410612305

### Additional context

Thereby, I've looked into the log and found this suggestion:
```console
ERROR: 10 leaked mock objects found at program exit. Expectations on a mock object are verified when the object is destructed. Leaking a mock means that its expectations aren't verified, which is usually a test bug. If you really intend to leak a mock, you can suppress this error using testing::Mock::AllowLeak(mock_object), or you may use a fake or stub instead of a mock.
```

Then, I've tried to add these lines in the branch `fix/windows_unittest`:
- https://github.com/robotology/icub-main/compare/master...fix/windows_unittest

Still unsuccessfully. See:
- https://github.com/robotology/icub-main/actions/workflows/ci.yml

cc @Nicogene

**Comment by pattacini** [comment]
Unittest temporarily disabled on Windows via https://github.com/robotology/icub-main/commit/629ec2bbc48548657f7acb78bdc4822a02d96cef, on `master` and `devel`.

The debug branch `fix/windows_unittest` has still the gtest enabled.

**Comment by pattacini** [comment]
The unittest makes use of specific releases of gtest, see:

https://github.com/robotology/icub-main/blob/629ec2bbc48548657f7acb78bdc4822a02d96cef/src/unittest/CMakeLists.txt#L15

which didn't change for a while[^1] and out of nowhere we started seeing failures since yesterday:
- https://github.com/robotology/icub-main/actions/runs/9410612305

[^1]: Actually, it was https://github.com/google/googletest/tree/release-1.12.0 but I updated it in the attempt to fix the problem, unsuccessfully. However v1.14.0 is still fairly old.

**Comment by traversaro** [comment]
I think the relevant change is the compiler (that is provide by GitHub Action image) that changed from to:

~~~
2024-06-06T05:51:18.0476112Z --   CMake: 3.29.3 (C:/Program Files/CMake/bin/cmake.exe)
2024-06-06T05:51:18.0478276Z --   C Compiler: MSVC 19.39.33523.0 (C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe)
2024-06-06T05:51:18.0481096Z --   CXX Compiler: MSVC 19.39.33523.0 (C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe)
~~~

to 

~~~
2024-06-07T05:26:38.4807555Z --   CMake: 3.29.3 (C:/Program Files/CMake/bin/cmake.exe)
2024-06-07T05:26:38.4809825Z --   C Compiler: MSVC 19.40.33811.0 (C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe)
2024-06-07T05:26:38.4812581Z --   CXX Compiler: MSVC 19.40.33811.0 (C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe)
~~~

**Comment by traversaro** [comment]
Relevant upstream issues:
* https://github.com/google/googletest/issues/4556
* https://github.com/actions/runner-images/issues/10004

**Comment by pattacini** [comment]
The problem has been solved upstream!
See:
- https://github.com/KhronosGroup/KTX-Software/pull/920

CI fully restored on `master` and `devel` via https://github.com/robotology/icub-main/commit/ab93283ba6d654396b92e081d960ace777a9e656.

**Comment by MarkCallow** [comment]
>The problem has been solved upstream! See:
>See:
>
>    https://github.com/KhronosGroup/KTX-Software/pull/920
>
>CI fully restored on master and devel via https://github.com/robotology/icub-main/commit/ab93283ba6d654396b92e081d960ace777a9e656.

I am thoroughly confused as to what a fix in KTX-Software has to do with this repo.

In the interests of accuracy, the referenced PR has nothing to do with fixing the SEH exceptions. The fix for that is the 2024.06.10.1.0 GHA runner image which makes sure `C:\Windows\system32` has latest vcruntime140.dll.

The referenced PR fixes a JVM crash when testing the Java binding caused by the JVM continuing to use an old `vcruntime140.dll`. It also fixes a breakage caused by a change in 2024.06.10 in how GitHub Actions handles setting environment variables from matrix attributes.

**Comment by pattacini** [comment]
Thanks for pointing this out, @MarkCallow 👍🏻 

I've merely jumped to the [final comment](https://github.com/actions/runner-images/issues/10004#issuecomment-2164776121) by @ijunaidm, who was asking the OP if everything was ok with the latest deployment.

The correct PR's fixing the SWH exception should be then:
- https://github.com/Mause/duckdb-java/pull/6
- https://github.com/duckdb/duckdb-java/pull/32
